File: internal/adapters/guest/fixtures/jvm-java-spring/src/test/java/dev/mds/GreetingControllerTest.java
package dev.mds;

import static org.junit.jupiter.api.Assertions.assertEquals;
import org.junit.jupiter.api.Test;

class GreetingControllerTest {
    @Test
    void returnsConfiguredGreeting() throws Exception {
        var controller = new GreetingController();
        var field = GreetingController.class.getDeclaredField("greeting");
        field.setAccessible(true);
        field.set(controller, "hello-java");
        assertEquals("hello-java", controller.greeting());
    }

    @Test
    void breakpointProbe() throws Exception {
        var controller = new GreetingController();
        var field = GreetingController.class.getDeclaredField("greeting");
        field.setAccessible(true);
        field.set(controller, "debug-java-test");
        assertEquals("debug-java-test", controller.greeting());
    }
}

File: internal/adapters/guest/fixtures/jvm-kotlin-spring/src/test/kotlin/dev/mds/GreetingControllerTest.kt
package dev.mds

import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.Test

class GreetingControllerTest {
    @Test
    fun returnsConfiguredGreeting() {
        assertEquals("hello-kotlin", GreetingController("hello-kotlin").greeting())
    }

    @Test
    fun breakpointProbe() {
        val controller = GreetingController("debug-kotlin-test")
        assertEquals("debug-kotlin-test", controller.greeting())
    }
}
